sdk/eventhub/Azure.Messaging.EventHubs/src/AzureClientBuilderExtensions.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Extensions;
+using Azure.Messaging.EventHubs;
+
+namespace Azure.ApplicationModel.Configuration
+{
+    /// <summary>
+    /// Extension methods to add <see cref="EventHubClient"/> client to clients builder
+    /// </summary>
+    public static class AzureClientBuilderExtensions
+    {
+        /// <summary>
+        /// Registers a <see cref="EventHubClient"/> instance with the provided <paramref name="connectionString"/>
+        /// </summary>
+        public static IAzureClientBuilder<EventHubClient, EventHubClientOptions> AddEventHubClient<TBuilder>(this TBuilder builder, string connectionString)
+            where TBuilder: IAzureClientFactoryBuilder
+        {
+            return builder.RegisterClientFactory<EventHubClient, EventHubClientOptions>(options => new EventHubClient(connectionString, options));
+        }
+
+        /// <summary>
+        /// Registers a <see cref="EventHubClient"/> instance with the provided <paramref name="connectionString"/> and <paramref name="eventHubPath"/>
+        /// </summary>
+        public static IAzureClientBuilder<EventHubClient, EventHubClientOptions> AddEventHubClient<TBuilder>(this TBuilder builder, string connectionString, string eventHubPath)
```

Comment:
I may not have context enough to be reading this entirely correctly, but I believe this limits scenarios to:

- Here's a connection string (with path embedded) and a set of options
- Here's a connection string and an event hub path, but you have to use default options  _(I suspect this will be the majority case)_

If that is the case, I'd suggest that we also allow:

- Here's a connection string, event hub path, and set of options

Reply:
`IAzureClientBuilder` return type has a ConfigureOptions(options => {}) method on it.

So to add a client and configure its options:

``` C#
builder.AddEventHubClient("connectionString", "path").ConfigureOptions(options => {} );
```

Reply:
👍    Works for me!